tests/integration/test_tls.py

```diff
@@ -47,7 +47,14 @@ async def test_deploy_tls(ops_test: OpsTest, kafka_charm):
     await asyncio.gather(
         ops_test.model.deploy(TLS_NAME, channel="beta", config=tls_config, series="jammy"),
         ops_test.model.deploy(ZK, channel="edge", series="jammy", application_name=ZK),
-        ops_test.model.deploy(kafka_charm, application_name=CHARM_KEY, series="jammy"),
+        ops_test.model.deploy(
+            kafka_charm,
+            application_name=CHARM_KEY,
+            series="jammy",
+            config={
+                "ssl_principal_mapping_rules": "RULE:^.*[Cc][Nn]=([a-zA-Z0-9.]*).*$/$1/L,DEFAULT"
+            },
```

Comment:
**praise:** Really good catch. I thought it defaulted to the CN, looked at the docs apparently its full DN.